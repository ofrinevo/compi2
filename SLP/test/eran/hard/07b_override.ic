/* 	This Program includes an proper method override of
    a method of one class in its sub-class. */ 
	
class A {
	
	int func ( int a ) { 
		return a + a ; 
	}

	static void main (string[] args) { 
		A b = new B();
		Library.println("b.func(37) =");
		Library.printi(b.func(37));
		Library.println("");
	}
}

class B extends A { 
	
	// overriding of func
	int func ( int a ) {
		return a + a + a ; 
	}
	
}
