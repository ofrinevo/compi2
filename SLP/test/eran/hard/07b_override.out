b.func(37) =
111
